Fix for issue #6519 (Pretty URLs in tests)

FoxCie opened the pull request:
When using pretty URLs in tests, the AdminUrlGenerator fails to provide the pretty URLs because it checks the use of pretty URLs through the context, which is null when doing requests from tests. This fixes this behaviour by checking the use of pretty URLs directly from the UrlGenerator (Router), which is what is done by the context anyway.

Tests have been adapted to use pretty URLs.

<!--
Thanks for your contribution! If you are proposing a new feature that is complex,
please open an issue first so we can discuss about it.

Note: all your contributions adhere implicitly to the MIT license
-->

javiereguiluz commented:
@FoxCie I'm not sure about these proposed changes.

Currently, we have separate tests for ugly URLs and pretty URLs. See https://github.com/EasyCorp/EasyAdminBundle/blob/c4d72d40a5da6cca67aa0e5dc6346a81d56d10b4/.github/workflows/ci.yaml#L146-L161

I prefer to have those two test suites separate because mixing pretty URLs with the existing tests that used the traditional ugly URLs is a mess in a my opinion. I'd prefer to keep both separated and then updated all tests to use pretty URLs only in the 5.x branch, which I hope to start soon.

FoxCie commented:
Hello @javiereguiluz

I reverted the changes to the tests, and added one in the PrettyUrlControllerTest to show that the AdminUrlGenerator does not behave well in tests when using pretty urls prior to this patch. This is not related to any controller, but since tests are invoked with this class only I added it there.

javiereguiluz commented:
Thanks a lot for fixing this bug, providing new tests ... and for your patience and the extra effort during the review process. This is merged now!
